horace_core/sqw/@rundatah/private/calc_sqw_.m

```diff
@@ -109,4 +111,14 @@
     obj.en,uoffset,  u_to_rlu, ...
     ulen,sqw_data.label,obj.run_id);
 
-header = Experiment([],obj.instrument,obj.sample,expdata);
\ No newline at end of file
+detpar = obj.det_par;
+
+if isempty(detpar)
+    header = Experiment([],obj.instrument,obj.sample,expdata);
```

Comment:
Below, these are `IX_detector_array()` rather than `[]`. Are these the same?

Reply:
(1) I tried IX_detector_array but at the moment it is not working, need some refactoring here, [] works
(2) Below is a unique_references_container containing the IX_detector_array, which does work